plugin/trino-apache-ranger/src/test/java/io/trino/plugin/ranger/TestRangerSystemAccessControl.java

```diff
@@ -0,0 +1,337 @@
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.function.SchemaFunctionName;
+import io.trino.spi.security.AccessDeniedException;
+import io.trino.spi.security.BasicPrincipal;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.SystemSecurityContext;
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.spi.security.ViewExpression;
+import io.trino.spi.type.VarcharType;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.security.auth.kerberos.KerberosPrincipal;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.trino.spi.security.PrincipalType.USER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestRangerSystemAccessControl
+{
+    static RangerSystemAccessControl accessControlManager;
+
+    private static final Identity ALICE = new Identity.Builder("alice").withPrincipal(new BasicPrincipal("alice")).build();
+    private static final Identity ADMIN = new Identity.Builder("admin").withPrincipal(new BasicPrincipal("admin")).build();
+    private static final Identity KERBEROS_INVALID_ALICE = Identity.from(ALICE).withPrincipal(new KerberosPrincipal("mallory/example.com@EXAMPLE.COM")).build();
+    private static final Identity BOB = Identity.ofUser("bob");
+
+    private static final Set<String> ALL_CATALOGS = ImmutableSet.of("open-to-all", "all-allowed", "alice-catalog", "user-catalog");
+    private static final Set<Identity> QUERY_OWNERS = ImmutableSet.of(Identity.ofUser("bob"), Identity.ofUser("alice"), Identity.ofUser("frank"));
+    private static final String CATALOG_ALICE = "alice-catalog";
+    private static final String CATALOG_USER_HOME = "user-catalog";
+    private static final CatalogSchemaName SCHEMA_ALICE_SCH1 = new CatalogSchemaName(CATALOG_ALICE, "sch1");
+    private static final CatalogSchemaTableName TABLE_ALICE_SCH1_TBL1 = new CatalogSchemaTableName(CATALOG_ALICE, "sch1", "tbl1");
+    private static final CatalogSchemaTableName VIEW_ALICE_SCH1_VW1 = new CatalogSchemaTableName(CATALOG_ALICE, "sch1", "vw1");
+    private static final CatalogSchemaRoutineName PROC_ALICE_SCH1_PROC1 = new CatalogSchemaRoutineName(CATALOG_ALICE, "sch1", "proc1");
+    private static final CatalogSchemaRoutineName FUNC_ALICE_SCH1_FUNC1 = new CatalogSchemaRoutineName(CATALOG_ALICE, "sch1", "func1");
+    private static final CatalogSchemaName SCHEMA_USER_BOB = new CatalogSchemaName(CATALOG_USER_HOME, "bob_schema");
+    private static final CatalogSchemaTableName TABLE_USER_BOB_TBL1 = new CatalogSchemaTableName(CATALOG_USER_HOME, SCHEMA_USER_BOB.getSchemaName(), "tbl1");
+
+    @BeforeAll
+    public static void setUpBeforeClass()
+            throws Exception
+    {
+        RangerConfig config = new RangerConfig();
```

Comment:
How do you configure `Ranger` here? I see you do not set what files should be read.

Don't you need to reference `plugin/trino-apache-ranger/src/test/resources/ranger-trino-security.xml` here?

Reply:
Ranger plugin library implicitly loads configurations from following files in the classpath, if found.
- ranger-trino-audit.xml
- ranger-trino-security.xml
- ranger-trino-policymgr-ssl.xml
 
Ranger test setup (EnvMultinodeApacheRangerextendEnvironment()) copies above files into Trino configuration directory.